addons/binding/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/MagentaTVLogger.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.magentatv.internal;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MagentaTVLogger} class implements a wrapper for slf4j.Logger and
+ * inserts a prefix to every messages, which makes it easier to read between all
+ * the other messages.
+ *
+ * @author Markus Michels - Initial contribution (markus7017)
+ */
+public class MagentaTVLogger {
```

Comment:
Ehm.. no. I know that openHABs way to communicate with the user can be limited (we only have the thing status so far), but that is not a reason to make the log file the ultimate solution for problem finding. Instead:
* Write self-healing code
* Use the thing status for everything else.
* Let exceptions escape to the framework. The framework will take the correct measures and in future might be able to communicate to the user via notifications or whatever.

Please remove this prefixer. You can also instruct slf4j via config files to prepend/append/hide/show for every logger instance

Reply:
don't agree 
- actually there is no user interaction beside logfile and thing status
- thing status is only viewable in PaperUI and you need to get the idea to look there
- as a developer I request the log from the user, which contains way more information that the thing status (even an update to that is not written to the log)
- slf4j: that would be manually requested to the user :-(
- catching the exception allows to output additional details, which are not included in the Exception object or the stack trace
- I'm very interested to understand how the framework should create a user notification based on the stacktrace and exception object - even then, it's not here, but I need those information to locate errors or understand what happened before...

Reply:
You forgot one thing: In one year you might not care about this binding anymore and someone else need to take over. Every line of code, every class adds to the complexity and reduces the maintainability.

This class is for pure aesthetic of log lines, you will get all info from the original log as well.